RELEASE-NOTES.md

```diff
@@ -5,7 +5,8 @@
 ### Enhancements
 
 - Added support for handling failed logins due to having an unverified account email [#2976](https://github.com/automattic/simplenote-electron/pull/2976)
-- Add support for handling passwords found in known data breaches [#2972](https://github.com/automattic/simplenote-electron/pull/2972)
+- Added support for handling passwords found in known data breaches [#2972](https://github.com/automattic/simplenote-electron/pull/2972)
+- Updated Simperium library dependency to latest version [#2984](https://github.com/automattic/simplenote-electron/pull/2984)
```

Comment:
@sandymcfadden I just wanted to double check that you'd like to include this in the user facing release notes. I'll add them during submission since you included them, but this seems more of an internal note to me. I am not sure if we can change the release notes before releasing on Monday, but we can chat depending on what you want to do about them.

Reply:
We can leave it internal like the other platforms:

https://github.com/Automattic/simplenote-ios/blob/develop/RELEASE-NOTES.txt#L10
https://github.com/Automattic/simplenote-android/blob/develop/RELEASE-NOTES.txt#L3
https://github.com/Automattic/simplenote-macos/blob/develop/RELEASE-NOTES.txt#L9

Reply:
While reviewing this PR, I noticed that other platforms also use the `[Internal]` prefix. I opened https://github.com/Automattic/simplenote-electron/pull/2990 to add that, but I don't consider it a blocker for approving this PR. 

Also, because it's only a change to the `RELEASE-NOTES.md`, I think it can be merged after the tag. But, if you'd rather have it straight to `develop`, feel free to take over the PR @oguzkocer

Reply:
That PR can target `develop` since it has no impact on this PR.